### PR TITLE
OED support for multi-currencies

### PR DESCRIPTION
**Pull request by sstruzik**
### Add multi-currency support for OED files
check if oed files contain multiple currencies. If they do then a currency converter is needed.
provide 3 ways for the user to pass a currency converter:
- via csv or parquet file
- using forex-python (need to be installed)
- by providing a path to your own module and class
To be set  are set via --currency-settings

### CODE REVIEW

**Review by sambles — COMMENTED** [review]
could be worth adding the `forex_python` package as an optional dependency in 
**optional-package.in**

**Review by sambles — CHANGES_REQUESTED** [review]
Update ods_tools minimum package version to the release which supports currency conversion
https://github.com/OasisLMF/OasisLMF/blob/7a2133c8a842391986ac949ad95fc2d53bdc25cd/requirements-package.in#L10

That way if the oasislmf package is updated a compatible version of ods_tools is also installed

**Review by sambles — CHANGES_REQUESTED** [review]
## multi-currency file - no conversion requested
When a multi-currency location file is given, but no conversion requested. The file generation will fail with 

```
$ oasislmf model generate-oasis-files
Warning: Unknown options(s) set in MDK config:
   analysis_settings_json : analysis_settings.json
   ktools_alloc_rule_gul : 1
   modelpy : True
   gulpy : True
   model_data_dir : model_data/PiWind

Stating oasislmf command - GenerateOasisFiles
RUNNING: oasislmf.manager.interface
  0%|                                                                                                                                                                                                                                                                                                                                                | 0/1 [00:00<?, ?it/s]
Processing arguments - Creating Oasis Files

Generating Oasis files (GUL=True, IL=True, RIL=True)
  0%|                                                                                                                                                                                                                                                                                                                                                | 0/1 [00:00<?, ?it/s]
file /home/sam/repos/models/piwind/runs/files-20220906083541/location.csv contains multiple currencies,reporting_currency and currency_conversion_json are necessary to perform computation
```  

## single currency file - conversion requested with missing data 
A similar but opposite case happens, where a single currency location file is given with a selected `--reporting-currency` but missing the required conversion data.
The oasis files generated, but without running a conversion or throwing a warning / exception 
```
$ oasislmf model generate-oasis-files --reporting-currency EUR 
Warning: Unknown options(s) set in MDK config:
   modelpy : True
   model_data_dir : model_data/PiWind
   ktools_alloc_rule_gul : 1
   gulpy : True
   analysis_settings_json : analysis_settings.json

Stating oasislmf command - GenerateOasisFiles
RUNNING: oasislmf.manager.interface
  0%|                                                                                                                                                                                                                                                                                                                                                | 0/1 [00:00<?, ?it/s]
Processing arguments - Creating Oasis Files

Generating Oasis files (GUL=True, IL=True, RIL=True)
RUNNING: oasislmf.lookup.factory.generate_key_files
COMPLETED: oasislmf.lookup.factory.generate_key_files in 0.19s
Keys successful: /home/sam/repos/models/piwind/runs/files-20220906084124/keys.csv generated with 40 items
Keys errors: /home/sam/repos/models/piwind/runs/files-20220906084124/keys-errors.csv generated with 20 items
   
   ... 
   
   $ cat /home/sam/repos/models/piwind/runs/files-20220906084124/location.csv | head -2
PortNumber,AccNumber,LocNumber,IsTenant,BuildingID,CountryCode,Latitude,Longitude,StreetAddress,PostalCode,OccupancyCode,ConstructionCode,LocPerilsCovered,BuildingTIV,OtherTIV,ContentsTIV,BITIV,LocCurrency,OEDVersion
1,A11111,10002082046,1,1,GB,52.76698052,-0.895469856,1 ABINGDON ROAD,LE13 0HL,1050,5000,WW1,220000,0,0,0,GBP,2.0.0


```

File: docs/OED_currency_support.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+if a currency pair is missing ValueError(f"currency pair {(cur_from, cur_to)} is missing") is thrown
+
+
+### FxCurrencyRates
+OasisLMF let you use the external package [forex-python](https://forex-python.readthedocs.io/en/latest/usage.html)
+to perform the conversion. A date may be specified in ISO 8601 format (YYYY-MM-DD)
+currency_conversion_json: 
+```json
+{
```
Comment (sambles) [inline docs/OED_currency_support.md:70]: Minor doc update, i guess the example should be 

```
{                                                                                                                                                                                                                                                                                                                                                                          
  "currency_conversion_type": "FxCurrencyRates",
  "datetime": "2018-10-10"
}
```

**Review by sambles — CHANGES_REQUESTED** [review]

File: docs/OED_currency_support.md (hunk 1 above)
Comment (sstruzik) [reply docs/OED_currency_support.md:70]: done

**Review by sambles — APPROVED** [review]
Looks good, thanks for the changes

**Pull request merged**